daemon/cmd/daemon_main.go

```diff
@@ -253,9 +253,17 @@ func init() {
 	flags.StringSlice(option.DebugVerbose, []string{}, "List of enabled verbose debug groups")
 	option.BindEnv(option.DebugVerbose)
 
-	flags.StringSliceP(option.Device, "d", []string{}, "List of devices facing cluster/external network for attaching bpf_netdev (first device should be one used for direct routing if tunneling is disabled)")
+	flags.StringP(option.Device, "d", "",
+		fmt.Sprintf("Device facing cluster/external network for attaching bpf_host. Deprecated in favor of --%s", option.Devices))
+	flags.MarkDeprecated(option.Device, "This option will be removed in v1.10")
 	option.BindEnv(option.Device)
```

Comment:
Can't we remove this right here and describe in the upgrade notes what a user needs to do? Why wait until 1.10?

Reply:
We would break the existing deployments in which `--device` was specified. I thought that we have a policy that we can change a user facing flag in a second release after we have announced it? @aanm might know better.

Reply:
correct. We can't simply remove flags and hope for the best :-)

Reply:
> correct. We can't simply remove flags and hope for the best :-)

But we eventually are going to remove them. I don't think anyone will read the daemon help. So it's basically the same issue in the second release from now.

Reply:
> > correct. We can't simply remove flags and hope for the best :-)
> 
> But we eventually are going to remove them. I don't think anyone will read the daemon help. So it's basically the same issue in the second release from now.

If the users use a deprecated flag it will show up in the logs also Martynas will add some notes in the upgrade guide.